### PR TITLE
JSUI-2709 [DynamicHierarchicalFacet] Add analytics and reduce the differences of implementation with DynamicFacet

### PR DESCRIPTION
**Pull request by ThibodeauJF**
Renamed clear/reset in order to fit the DynamicFacet's public methods. Would be confusing otherwise. Previously had to follow what the CategoryFacet was doing

[Deploy](https://dashboard.heroku.com/pipelines/a3535101-5bbf-4a5b-a909-47fcf8c9f149)

### CODE REVIEW

File: src/ui/DynamicFacet/DynamicFacet.ts [hunk 1]
```diff
@@ -763,7 +764,7 @@ export class DynamicFacet extends Component implements IDynamicFacet {
   }
 
   private logAnalyticsFacetShowMoreLess(cause: IAnalyticsActionCause) {
-    this.usageAnalytics.logCustomEvent<IAnalyticsFacetState>(cause, this.basicAnalyticsFacetState, this.element);
+    this.usageAnalytics.logCustomEvent<IAnalyticsFacetMeta>(cause, this.basicAnalyticsFacetMeta, this.element);
```
Comment (ThibodeauJF) [inline src/ui/DynamicFacet/DynamicFacet.ts]: that was a oopsie

File: src/ui/DynamicHierarchicalFacet/DynamicHierarchicalFacet.ts [hunk 2]
```diff
@@ -280,6 +284,13 @@ export class DynamicHierarchicalFacet extends Component implements IDynamicHiera
     Assert.exists(data);
     Assert.exists(data.queryBuilder);
     this.putStateIntoQueryBuilder(data.queryBuilder);
+    this.putStateIntoAnalytics();
+  }
+
+  private verifyCollapsingConfiguration() {
+    if (this.options.collapsedByDefault && !this.options.enableCollapse) {
+      this.logger.warn('The "collapsedByDefault" option is "true" while the "enableCollapse" is "false"');
```
Comment (fbeaudoincoveo) [inline src/ui/DynamicHierarchicalFacet/DynamicHierarchicalFacet.ts]: Suggestion:

> Setting "collapseByDefault" to "true" has no effect when "enableCollapse" is set to "false"

File: unitTests/ui/DynamicHierarchicalFacet/DynamicHierarchicalFacetTestUtils.ts [hunk 3]
```diff
@@ -103,6 +104,15 @@ export class DynamicHierarchicalFacetTestUtils {
     facet.values = new DynamicHierarchicalFacetValues(facet);
     facet.element = $$('div').el;
     facet.searchInterface = Mock.mockSearchInterface();
+    facet.header = Mock.mock(DynamicFacetHeader);
+    facet.header.options = {
```
[the remark this answers is not in the document]
Comment (ThibodeauJF) [reply unitTests/ui/DynamicHierarchicalFacet/DynamicHierarchicalFacetTestUtils.ts]: It's smart enough to recognize the IDynamicFacetHeaderOptions type and validate it 😄

File: src/ui/DynamicFacet/DynamicFacet.ts [hunk 4]
```diff
@@ -618,14 +619,10 @@ export class DynamicFacet extends Component implements IDynamicFacet {
 
     if (Utils.isNonEmptyArray(valuesToSelect)) {
       this.selectMultipleValues(valuesToSelect);
-      // Only one search event is sent, pick first facet value
```
Comment (ThibodeauJF) [inline src/ui/DynamicFacet/DynamicFacet.ts]: Those weren't even sent, the HistoryController took priority

File: src/ui/DynamicFacet/DynamicFacet.ts [hunk 5]
```diff
@@ -579,19 +581,18 @@ export class DynamicFacet extends Component implements IDynamicFacet {
...
 
-  private onQueryResponse(response?: IFacetResponse) {
-    if (response) {
-      this.moreValuesAvailable = response.moreValuesAvailable;
-      return this.values.createFromResponse(response);
-    }
+  private onQueryResponse(response: IFacetResponse) {
+    this.moreValuesAvailable = response.moreValuesAvailable;
+    this.values.createFromResponse(response);
+  }
```
Comment (rdeandaCoveo) [inline src/ui/DynamicFacet/DynamicFacet.ts]: Maybe something like `onValuesFound` could be more specific now that you've extracted the condition to the caller function.

Comment (ThibodeauJF) [reply src/ui/DynamicFacet/DynamicFacet.ts]: still it's receiving the query response as parameter and then builds the values from it

Comment (rdeandaCoveo) [reply src/ui/DynamicFacet/DynamicFacet.ts]: As both functions are being called in the same ternary, it seems to me the naming of them must be at the same level of abstraction. `onQueryResponse` sounds more like plumbing related, and `onNoValues` sounds more business or domain related.

But no big deal ;-)

**Pull request merged**